asst1/CODE/my_scheduler.c

```diff
@@ -111,19 +111,40 @@ void run_maintenance() {
       if (new_prio == thread->priority) { // do nothing
         prev_ptr = ptr;
         ptr = ptr->next;
-      } else { // remove thread from current queue and add it to lower queue
+      } 
+      else { // remove thread from current queue and add it to lower queue
         thread->priority = new_prio;
         insert_ready_q(thread, new_prio);
         if (ptr == ready_q[i]->head) {
           ready_q[i]->head = ptr->next;
-        } else {
+        } 
+	else {
           prev_ptr->next = ptr->next;
         }
         if (ptr == ready_q[i]->tail) {
-          ready_q[i]->tail = NULL;
+          ready_q[i]->tail = prev_ptr;
         }
+	node_t* temp = ptr;
         ptr = ptr->next;
+	free(temp);
+	fCount++;
       }
     }
   }
 }
+
+	
+void free_data() {
+  //  printf("here\n");
+    tcb* thread = get(all_threads, 2);
+    free_list(thread->waited_on);
+    free(thread->context.uc_stack.ss_sp);
```

Comment:
and if `thread` is the main thread, there's nothing to free since we don't alloc stack